presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -521,6 +522,25 @@ public static Slice stGeometryN(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlTy
         return serialize(ogcGeometry);
     }
 
+    @SqlNullable
+    @Description("Returns the Point geometry element at the specified index (indices started with 1)")
+    @ScalarFunction("ST_PointN")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stPointN(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlType(INTEGER) long n)
+    {
+        OGCGeometry geometry = deserialize(input);
+        if (GeometryType.getForEsriGeometryType(geometry.geometryType()) != LINE_STRING) {
+            return null;
```

Comment:
This function should fail if input type is not linestring.

`validateType("ST_PointN", geometry, EnumSet.of(LINE_STRING));`